### ISSUE TITLE
Use enums for Hibernate ORM configuration

### ISSUE THREAD

**Issue by gsmet** [quarkusio/quarkus#3945]
This will allow to have the values properly documented.

**Comment by Sanne** [comment]
@gsmet still plan to do this?

**Comment by gastaldi** [comment]
What configuration are you referring to in this issue exactly?

**Comment by gsmet** [comment]
Typically `quarkus.hibernate-orm.database.generation`.

**Comment by gsmet** [comment]
Note that I'm not sure it's worth the effort if we don't have enums on the ORM side.

**Comment by gastaldi** [comment]
Yeah, I agree, perhaps it would be nice to have an issue in the Hibernate JIRA asking for this

**Comment by yrodiere** [comment]
I think a problem with having enums on the Hibernate ORM side -- which @gsmet mentioned to me recently in the context of Hibernate Search -- is that we wouldn't get javadoc-extracted documentation for enum values, so wouldn't have any tooltip explaining what each value means.

Perhaps not a deal-breaker, but something to keep in mind.

**Comment by yrodiere** [comment]
A few pointers (cc @jrenaat):

* This would involve changing the return type of some methods in "configuration classes" -- the classes that "map" configuration coming from application.properties.
* The relevant configuration classes for Hibernate ORM are `HibernateOrmConfig`, `HibernateOrmConfigPersistenceUnit`, `HibernateOrmRuntimeConfig`, `HibernateOrmRuntimeConfigPersistenceUnit`
* We want to change the type of methods returning `String`, `List<String>`, `Optional<String>` to use an enum instead of a string
* You can see an example using an enum here: https://quarkus.io/guides/hibernate-search-orm-elasticsearch#quarkus-hibernate-search-orm-elasticsearch_quarkus-hibernate-search-orm-elasticsearch-protocol , corresponding code: https://github.com/quarkusio/quarkus/blob/821a9afefcb8dbb65aa4898034b8e10c3e2f1c94/extensions/hibernate-search-backend-elasticsearch-common/runtime/src/main/java/io/quarkus/hibernate/search/backend/elasticsearch/common/runtime/HibernateSearchBackendElasticsearchRuntimeConfig.java#L34
* See how docs automatically display the list of possible values, with a tooltip on hover explaining what each value does.
* The tooltip content is automatically extracted from javadoc on each enum value -- but I think that only works if the enum is in Quarkus itself, so we might want to duplicate some enums.
* Of course after changing the return type of a method, we'll need to adjust the call sites to convert it back to a string, or to an enum defined in Hibernate ORM where possible. Not great, but this is what it takes.
* To build Quarkus and especially the docs, see https://github.com/quarkusio/quarkus/blob/main/CONTRIBUTING.md#documentation

HTH, let's talk if not
